lib/sandbox/timers.js

```diff
@@ -158,11 +157,15 @@ function Timerz (delegations, onError, onAnyTimerStart, onAllTimerEnd) {
     computeTimerEvents = function (increment, clearing) {
         increment && (pending += increment);
 
-        if (pending === 0 && computeTimerEvents.started) {
-            !clearing && (typeof onAllTimerEnd === FUNCTION) && onAllTimerEnd();
-            computeTimerEvents.started = false;
+        function maybeEndAllTimers () {
+            if (pending === 0 && computeTimerEvents.started) {
```

Comment:
This looks like a duplicate check from line 167.

Reply:
@codenirvana The first check is "only queue this if `pending` is currently at 0."  The second check is "only actually run this if `pending` is still at 0."  This is the one we really need.  The first check just prevents needless queueing.